src/popup/components/TransactionList.vue

```diff
@@ -73,133 +63,150 @@ export default {
+    const activeNetwork = useGetter('activeNetwork');
+    const getAccountPendingTransactions = useGetter('getAccountPendingTransactions');
+    const getTxSymbol = useGetter('getTxSymbol');
+
+    const compareCaseInsensitive = (
+      str1: string,
+      str2: string,
+    ) => str1.toLocaleLowerCase() === str2.toLocaleLowerCase();
+
+    const filteredTransactions = computed(() => {
+      const isFungibleTokenTx = (tr: any) => Object.keys(availableTokens.value)
+        .includes(tr.tx.contractId);
+
+      return [...transactions.value.loaded, ...getAccountPendingTransactions.value]
+        .filter((tr) => (!props.token
+          || (props.token !== AETERNITY_CONTRACT_ID
+            ? tr.tx?.contractId === props.token
+            : (!tr.tx.contractId
+              || !isFungibleTokenTx(tr)))))
+        .filter((tr) => {
+          switch (displayMode.value.filter) {
+            case 'all':
+              return true;
+            case 'dex':
+              return getDexContracts.value && tr.tx.contractId && (
+                getDexContracts.value.router.includes(tr.tx.contractId)
+                || getDexContracts.value.wae?.includes(tr.tx.contractId)
+              );
+            case 'out':
+              return (compareCaseInsensitive(tr.tx.type, SCHEMA.TX_TYPE.spend)
+                  && tr.tx.senderId === account.value.address)
+                || (isFungibleTokenTx(tr)
+                  && compareCaseInsensitive(tr.tx.type, SCHEMA.TX_TYPE.contractCall)
+                  && tr.tx.callerId === account.value.address);
+            case 'in':
+              return (compareCaseInsensitive(tr.tx.type, SCHEMA.TX_TYPE.spend)
+                  && (tr.tx.recipientId === account.value.address || (tr.tx.senderId !== account.value.address && tr.tx.recipientId.startsWith('nm_'))))
+                || (isFungibleTokenTx(tr)
+                  && compareCaseInsensitive(tr.tx.type, SCHEMA.TX_TYPE.contractCall)
+                  && tr.recipient === account.value.address);
+            case 'tips':
+              return (tr.tx.contractId
+                && (activeNetwork.value.tipContractV1 === tr.tx.contractId
+                  || activeNetwork.value.tipContractV2 === tr.tx.contractId)
+                && (tr.tx.function === 'tip' || tr.tx.function === 'retip')) || tr.claim;
+            default:
+              throw new Error(`${root.$t('pages.recentTransactions.unknownMode')} ${displayMode.value.filter}`);
+          }
+        })
+        .filter(
+          (tr) => !searchBar.value
+            || getTxSymbol.value(tr)
+              .toLocaleLowerCase()
+              .includes(searchBar.value.toLocaleLowerCase()),
+        )
+        .sort((a, b) => {
+          const arr: any[] = [a, b].map((tr) => new Date(tr.microTime));
+          if (displayMode.value.rotated) arr.reverse();
+
+          return (arr[0] - arr[1]) || a.pending;
+        })
+        .slice(0, props.maxLength || Infinity);
```

Comment:
I think you should consider splitting that code into separate functions to make it readable

Reply:
https://aeternity-blockchain.atlassian.net/browse/SW-533